File: c/problem13.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

// Problem 13 for Project Euler
// https://projecteuler.net/problem=13

int main(void)
{
	char list[100][50];
	FILE *fp = fopen("number.txt", "r");
	int i = 0, k, temp = 0, num;

	if (fp == NULL){
		printf("Failed to open file\n");
		exit(1);
	}

	while (fscanf(fp, "%s", &list[i]) != EOF)
		i++;

	for (k = 49; k >= 0; k--)
	{
		num = 0;
		for (i = 0; i<=99 ; i++)
			num += (list[i][k]-'0');
		temp += num;
		printf("%d\n", temp % 10);
		if (temp != 0)
			temp /= 10;
	}
	fclose(fp);
	printf("Ans: %d\n", temp);

	return 0;
}

File: python/problem13.py
'''
Problem 13, https://projecteuler.net/problem=13

Work out the first ten digits of the sum of the following one-hundred 50-digit numbers.
See p13.txt for the number
'''

ans = 0;
number_file = open('p13.txt')
number_list = number_file.readlines()
for number in number_list:
	ans += int(number)

print str(ans)[:10]
# Sort of one liner?
print str(sum([int(x) for x in number_list]))[:10]